tests/utils/test_rfree.py

```diff
@@ -18,6 +18,24 @@ def test_add_rfree(self):
         self.assertFalse("R-free-flags" in data.columns)
         self.assertTrue("R-free-flags" in rfree.columns)
 
+        # Should keep all other unchanged
+        self.assertTrue(np.all(data == rfree.loc[:, rfree.columns != "R-free-flags"]))
+        # Should have fewer ==1 as test set
+        self.assertTrue(np.sum(rfree.loc[:, "R-free-flags"] == 1) / len(rfree) < 0.15)
+
+        # Test ccp4_convention
+        rfree_ccp4 = rs.utils.add_rfree(data, ccp4_convention=True)
+        self.assertTrue("FreeR_flag" in rfree_ccp4.columns)
+
+        # Should keep all other unchanged
+        self.assertTrue(
+            np.all(data == rfree_ccp4.loc[:, rfree_ccp4.columns != "FreeR_flag"])
+        )
+        # Should have fewer ==0 as test set
+        self.assertTrue(
+            np.sum(rfree_ccp4.loc[:, "FreeR_flag"] == 0) / len(rfree_ccp4) < 0.15
+        )
+
```

Comment:
This is all fine, but it would be useful to move this test over to `pytest`, rather than keeping it in `unittest` (this is purely this way for historical reasons). Specifically, this would allow us to `parametrize` over `ccp4_convention` and `fraction`.

Reply:
let me know if you're up for including that refactor in this PR -- otherwise I can tackle it in another one.

Reply:
Do you mean rewriting the rest function with `pytest.mark.parameterize` decorators? Sure, I can tackle it in this PR.

Reply:
yep -- using the `parametrize` decorators, and rewriting it to a function-based, pytest style, rather than the object-based `unittest` one. This move was made because we find the `pytest` style to be more concise and expressive, with less boiler plate than the `unittest` style.